components/canvas/utils/handlePasteNode.tsx

```diff
@@ -0,0 +1,37 @@
+import { NodeData, NodeDataFull } from "@/types/NodeData";
+import { NodeTypes } from "@/types/NodeTypes";
+import { Position, Node } from "reactflow";
+import { isValidTarget } from "./isValidTarget";
+import { NodeDataApiRequestBody } from "@/types/NodeDataApi";
+import { getNodeTypeName } from "@/utils/getNodeTypeName";
+
+export const handlePasteNode = (
+  source: Node<NodeData>,
+  target: Node<NodeData> | undefined,
+  nodes: Node<NodeDataFull, string | undefined>[],
+  addNode: (
+    parentId: string,
+    data: NodeDataApiRequestBody,
+    position: Position
+  ) => void
+) => {
+  const sourceType = getNodeTypeName(source.data.type);
+  const targetType = getNodeTypeName(target?.data.type);
+
+  if (!target?.id) {
+    throw new Error(
+      "Unable to paste: Hover a card before attempting to paste ⛔"
+    );
+  }
```

Comment:
Can you move this before line 18 and 19 to not have to open up for `undefined` target type in `getNodeTypeName`?